docs/data/date-pickers/custom-components/CalendarHeaderComponentRange.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { Dayjs } from 'dayjs';
+import { styled } from '@mui/material/styles';
+import Typography from '@mui/material/Typography';
+import IconButton from '@mui/material/IconButton';
+import ChevronLeft from '@mui/icons-material/ChevronLeft';
+import ChevronRight from '@mui/icons-material/ChevronRight';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateRangeCalendar } from '@mui/x-date-pickers-pro/DateRangeCalendar';
+import { PickersRangeCalendarHeaderProps } from '@mui/x-date-pickers-pro/PickersRangeCalendarHeader';
+
+const CustomCalendarHeaderRoot = styled('div')({
+  display: 'flex',
+  justifyContent: 'space-between',
+  padding: '8px 16px',
+  alignItems: 'center',
+});
+
+function CustomCalendarHeader(props: PickersRangeCalendarHeaderProps<Dayjs>) {
+  const { currentMonth, onMonthChange, month, calendars, monthIndex } = props;
+
+  const selectNextMonth = () =>
+    onMonthChange(currentMonth.add(calendars, 'month'), 'left');
+  const selectPreviousMonth = () =>
+    onMonthChange(currentMonth.subtract(calendars, 'month'), 'right');
+
+  return (
+    <CustomCalendarHeaderRoot>
+      <IconButton
```

Comment:
In yet another example?
Because for now I'm showcasing #10099

Reply:
WDYT about adding it to the same demo? 🤔

Reply:
Not a huge fan, because right now the demo has a very clear purpose: move the months 2 by 2
If we add the year switching, we are providing two example at once, but most people will probably either want one or the other (or something totally different of course)

Do you think that building #10566  is not trivial with the tools demonstrated in this demo?

Reply:
It does not seem hard from our PoV, but I fear that if we don't add it, people will be still opening issues and asking for it. 🙈 
Do you think that it would increase the complexity that much? 🤔

Reply:
I guess it's a balance 
It's not **that hard** to do it yourself and it does not increase the complexity **that much** :laughing: 

Maybe we could just use the `DateCalendar` demo to show the year switches, after all this request is not only legit to the range picker, whereas the multiple month switch kinda is.

Reply:
Fair point, that seems like a great compromise. 👌

Reply:
Done :+1:

Reply:
Superb examples! 💙 
But the new example clearly shows the problem with using only month as the transition key. 😱 🙈 

https://github.com/mui/mui-x/blob/b66a47275554caebbd61c3ad97a83defa6dab01c/packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx#L500C9-L500C22

Reply:
Good catch
Do you think we should just add the year?

Reply:
Yeah, I think that it would make sense to have a `${year}-${month}` as the key, WDYT? 🤔

Reply:
Make total sense, done